public/app/features/explore/RichHistory/RichHistoryQueriesTab.tsx

```diff
@@ -122,20 +124,22 @@ export function RichHistoryQueriesTab(props: RichHistoryQueriesTabProps) {
     clearRichHistoryResults,
     loadMoreRichHistory,
     richHistorySettings,
-    exploreId,
     height,
-    activeDatasourceInstance,
   } = props;
 
+  const exploreActiveDS = useSelector(selectExploreDSMaps);
   const styles = useStyles2(getStyles, height);
 
   const listOfDatasources = createDatasourcesList();
 
+  // on mount, set filter to either active datasource or all datasources
   useEffect(() => {
     const datasourceFilters =
-      !richHistorySettings.activeDatasourceOnly && richHistorySettings.lastUsedDatasourceFilters
+      !richHistorySettings.activeDatasourcesOnly && richHistorySettings.lastUsedDatasourceFilters
         ? richHistorySettings.lastUsedDatasourceFilters
-        : [activeDatasourceInstance];
+        : exploreActiveDS.dsToExplore
+            .map((eDs) => listOfDatasources.find((ds) => ds.uid === eDs.datasource?.uid)?.name)
+            .filter((name): name is string => !!name);
```

Comment:
Why do we need to filter out data sources without name?

Reply:
Because Rich history filters are by name, not UID https://github.com/grafana/grafana/blob/main/public/app/core/utils/richHistoryTypes.ts#L24